index.js

```diff
@@ -28,7 +28,21 @@ const exampleMovies = require("./movies");
       "James and the Giant Peach",
     ];
  */
-function getAllMovieTitles() {}
+/* input - array of movie objects [], number 
+output - new array of strings of movie titles
+plan - loop thru and find the movies[].title  and return all titles
+
+default error - if movie [] is empty return empty array
+*/
+function getAllMovieTitles(movies) {
+  let movieTitles = [];
+  if (!movies || !movies.length) {
+    movieTitles = [];
+  } for (let i = 0; i < movies.length; i++) {
+    movieTitles.push(movies[i].title);
+  }
+  return movieTitles;
+}
```

Comment:
This is great Jede - you've exceeded what was asked in the error handling and data validation.  Id move you `for` loop down to a new line to clean it up.  Also - while I _love_ to see your pseudo code we should get in the habit of removing it before submission moving forward.